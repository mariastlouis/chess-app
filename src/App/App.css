@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");
@import url("https://fonts.googleapis.com/css?family=Cabin");
body {
  font-family: "Cabin", sans-serif; }

.app {
  display: flex;
  flex-direction: row;
  justify-content: center; }

.game-board {
  max-width: 950px;
  width: 70%; }

.sider {
  max-width: 400px;
  margin: 0px auto;
  padding: 10px;
  width: 30%; }
