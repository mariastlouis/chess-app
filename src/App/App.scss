@import 'variable';

.app {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.game-board {
  max-width: 950px;
  width: 70%;

}

.sider {
  max-width: 400px;
  margin: 0px auto;
  padding: 10px;
  width: 30%;
}