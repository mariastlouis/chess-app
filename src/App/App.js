import React, { Component } from 'react';
import './App.css';
import Board from '../Board/Board';
import ControlPanel from '../ControlPanel/ControlPanel'


class App extends Component {
  constructor () {
    super();
    this.state = {
      knightPosition:[1,7],
      bishopPosition: [2, 7],
      selectedPiece: '',
      message: 'To move the pieces, first click on a piece and then click on the square you want it to move to. Note that the pieces can only make valid chess moves'
    };
    this.setSquare = this.setSquare.bind(this);
  }

  checkSelectedPiece(x, y) {
    const piece = this.state.selectedPiece;
    if (piece) {
      this.checkMoveValid(x, y)
    } else {
      this.setState({message: 'Select a piece to move'})
    }
  }

  checkMoveValid(x, y) {
    if (this.state.selectedPiece === 'knight') {
      this.moveKnight(x, y)
    } else {
      this.moveBishop(x,y)
    }
  }

  // if there is a piece selected && if the new x, y coordinate equals the same coordinate of the other piece, then you can't move


   checkSquareDuplicate(x,y) {
    const [knightX, knightY] = this.state.knightPosition;
    const [bishopX, bishopY] = this.state.bishopPosition;
    if((this.state.selectedPiece === 'knight') && (bishopX === x && bishopY === y)) {
      this.setState({message: 'Invalid move. Move to a space that is not occupied by the bishop'})
    } else if ((this.state.selectedPiece === 'bishop') && (knightX === x && knightY === y)){
        this.setState({message: 'Invalid move. Move to a space that is not occupied by the knight'})
    } else {
      this.checkMoveValid(x, y)
    }
  }



  moveKnight(x,y){
    const [knightX, knightY] = this.state.knightPosition
    const moveX = Math.abs(x - knightX);
    const moveY = Math.abs(y - knightY);

    if((moveX === 2 && moveY === 1) || (moveX === 1 && moveY === 2) ){

      this.setState({knightPosition: [x, y]})
      this.setState({message: 'Valid knight move. Select another piece to move'})
      this.setState({selectedPiece: ''})
    } else {
      this.setState({message: 'Invalid move -- knights can only move in an L shape'})
    }
  }

  moveBishop(x, y) {
    const [bishopX, bishopY] = this.state.bishopPosition
    const moveX = Math.abs(x - bishopX);
    const moveY = Math.abs(y - bishopY);

    if(moveX === 1 && moveY === 1) {
      this.setState({bishopPosition: [x, y]})
      this.setState({message: 'Valid bishop move. Select another piece to move'})
      this.setState({selectedPiece: ''})
    } else {
       this.setState({message: 'Invalid move - bishops can only move diagonally'})
    }
  }

  // setSquare(x,y) {
  //   if(piece) {

  //   }
  // }

// need to check to see if the piece you are clicking already contains the coordinates for the other piece
// error is happening when you are trying to switch over pieces. You have a knight selected and are trying to select a bishop piece. You only want the error on move to happen if it is a valid move. If it is not a valid move, it should proceed as normal. 
  // 1. click on a piece
  // 2. if the piece you clicked on equals the same coordinates as the knight or bishop piece, that piece is selected in the state
// 3. if it does not equal the coordinates - it then checks to see if a piece is already selected
// 4. if a piece is already selected - then it checks to see if the position you want to move to is a valid chess move
// 5. if it's valid, the piece moves. if it is not valid, the piece doesn't move and you get an error message
  
  // if this.state.selectedPiece == knight & xy coordinates you get === bishop coordinates then you can't move. and if the selected piece is a bishop and the xy coordinates === knight coordinates then you can't move


  setSquare(x, y) {
   
    const piece = this.state.selectedPiece;

    if(piece) {
      this.checkSquareDuplicate(x,y)
    } else {
      this.setSquare2(x,y)
    }
  }

  //   checkSquareDuplicate(x,y) {
  //   const [knightX, knightY] = this.state.knightPosition;
  //   const [bishopX, bishopY] = this.state.bishopPosition;
  //   if((this.state.selectedPiece === 'knight') && (bishopX === x && bishopY === y)) {
  //     this.setState({message: 'Invalid move. Move to a space that is not occupied by the bishop'})
  //   } else if ((this.state.selectedPiece === 'bishop') && (knightX === x && knightY === y)){
  //       this.setState({message: 'Invalid move. Move to a space that is not occupied by the knight'})
  //   } else {
  //     console.log('can move')
  //     this.checkMoveValid(x, y)
  //   }
  // }



  setSquare2 (x, y) {
    const [knightX, knightY] = this.state.knightPosition;
  
    const [bishopX, bishopY] = this.state.bishopPosition;
   



      this.setState({message: ''})
    if(knightX === x && knightY === y) {
      this.setState({selectedPiece: 'knight'})
    } else if (bishopX === x && bishopY === y ){
      this.setState({selectedPiece: 'bishop'})
    } else {
      this.checkSelectedPiece(x, y)
  }
}

  render() {
    return (
      <div className="app">
        <div className = "game-board">
          <Board 
          knightPosition={this.state.knightPosition}
          bishopPosition = {this.state.bishopPosition}
          setSquare = {this.setSquare}
          chessPiece = {this.state.selectedPiece}
          selectPiece = {this.selectPiece}/>
        </div>
        <div className = "sider">
          <ControlPanel message = {this.state.message}
                        chessPiece = {this.state.selectedPiece}/>
        </div>
      </div>
    );
  }
}

export default App;
