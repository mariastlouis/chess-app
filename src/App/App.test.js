import React from 'react';
import App from './App.js';
import {shallow} from 'enzyme';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

let renderedApp;

describe('App', () => {
  beforeEach(() => {
    renderedApp = shallow(
      <App />)
  });
  
  it('should render correctly', () => {
    expect(renderedApp).toBeDefined();
  });

  it('should match the snapshot', () =>{
    expect(renderedApp).toMatchSnapshot();
  });


});