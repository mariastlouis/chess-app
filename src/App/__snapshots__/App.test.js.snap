// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="sider"
>
          <ControlPanel
                    chessPiece=""
                    message="To move the pieces, first click on a piece and then click on the square you want it to move to. Note that the pieces can only make valid chess moves"
                    mode={[Function]}
          />
</div>,
        <div
          className="game-board"
>
          <Board
                    bishopPosition={
                              Array [
                                        2,
                                        7,
                                      ]
                    }
                    checkMoveValid={[Function]}
                    chessPiece=""
                    gameMode="standard"
                    knightPosition={
                              Array [
                                        1,
                                        7,
                                      ]
                    }
                    selectPiece={undefined}
                    setSquare={[Function]}
          />
</div>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ControlPanel
            chessPiece=""
            message="To move the pieces, first click on a piece and then click on the square you want it to move to. Note that the pieces can only make valid chess moves"
            mode={[Function]}
/>,
          "className": "sider",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "chessPiece": "",
            "message": "To move the pieces, first click on a piece and then click on the square you want it to move to. Note that the pieces can only make valid chess moves",
            "mode": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Board
            bishopPosition={
                        Array [
                                    2,
                                    7,
                                  ]
            }
            checkMoveValid={[Function]}
            chessPiece=""
            gameMode="standard"
            knightPosition={
                        Array [
                                    1,
                                    7,
                                  ]
            }
            selectPiece={undefined}
            setSquare={[Function]}
/>,
          "className": "game-board",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bishopPosition": Array [
              2,
              7,
            ],
            "checkMoveValid": [Function],
            "chessPiece": "",
            "gameMode": "standard",
            "knightPosition": Array [
              1,
              7,
            ],
            "selectPiece": undefined,
            "setSquare": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sider"
>
            <ControlPanel
                        chessPiece=""
                        message="To move the pieces, first click on a piece and then click on the square you want it to move to. Note that the pieces can only make valid chess moves"
                        mode={[Function]}
            />
</div>,
          <div
            className="game-board"
>
            <Board
                        bishopPosition={
                                    Array [
                                                2,
                                                7,
                                              ]
                        }
                        checkMoveValid={[Function]}
                        chessPiece=""
                        gameMode="standard"
                        knightPosition={
                                    Array [
                                                1,
                                                7,
                                              ]
                        }
                        selectPiece={undefined}
                        setSquare={[Function]}
            />
</div>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ControlPanel
              chessPiece=""
              message="To move the pieces, first click on a piece and then click on the square you want it to move to. Note that the pieces can only make valid chess moves"
              mode={[Function]}
/>,
            "className": "sider",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "chessPiece": "",
              "message": "To move the pieces, first click on a piece and then click on the square you want it to move to. Note that the pieces can only make valid chess moves",
              "mode": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Board
              bishopPosition={
                            Array [
                                          2,
                                          7,
                                        ]
              }
              checkMoveValid={[Function]}
              chessPiece=""
              gameMode="standard"
              knightPosition={
                            Array [
                                          1,
                                          7,
                                        ]
              }
              selectPiece={undefined}
              setSquare={[Function]}
/>,
            "className": "game-board",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "bishopPosition": Array [
                2,
                7,
              ],
              "checkMoveValid": [Function],
              "chessPiece": "",
              "gameMode": "standard",
              "knightPosition": Array [
                1,
                7,
              ],
              "selectPiece": undefined,
              "setSquare": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
